uvtools/dspec.py

```diff
@@ -1235,9 +1243,79 @@ def delay_filter_leastsq(data, flags, sigma, nmax, add_noise=False, freq_units =
     return mdl_array, cn_array, inp_data
 
 
-def sinc_downweight_mat_inv(nchan, df, filter_centers, filter_widths,
-                            filter_factors, cache = {}, wrap = False, wrap_interval=1,
-                            nwraps = 1000, no_regularization = False):
+def delay_interpolation_matrix(nchan, ndelay, wgts, fundamental_period=None, cache={}, taper='none', return_diagnostics=False):
+    """
+    Copute a foreground interpolation matrix.
+
+    Computes a foreground interpolation matrix that, when applied to data,
+    interpolates over flagged channels with delays between
+    -ndelay / fundamental_period, ndelay / fundamental_period
+
+    The computed Matrix is equal to F = A @ [ A^T @ W @ A]^{-1} @ A^T W
+    where A is an nchan \times 2ndelay  design matrix
+    y = A \tilde{y}
+    y is the frequency representation of data and \tilde{y} is
+    a 2xndelay vector holding the data's fourier coefficients. W is a diagonal
+    matrix of frequency-data weights. The net effect of F, when applied to flagged
+    data, is to solve for the fourier coefficients fitting unflagged channels
+    ([ A^T @ W @ A]^{-1} @ A^T W solves the linear least squares problem) and then return
+    the unflagged Fourier transform by apply A @ to the fitted coefficients, resulting
+    in data that is linearly interpolated.
+
+    Parameters
+    ----------
+    nchan: int
+        Number of frequency channels to interpolate over.
+    ndelay,
```

Comment:
Looks like the type is still missing here (should be `ndelay: int`).

Reply:
I've fixed this.